lib/sidebar/index.ts

```diff
@@ -0,0 +1,163 @@
+	 */
+	iconSvg: string
+
+	/**
+	 * The order of this tab.
+	 * Use a low number to make this tab ordered in front.
+	 */
+	order: number
+
+	/**
+	 * Callback to check if the sidebar tab should be shown for the selected node.
+	 *
+	 * @param node - The currently selected node
+	 */
+	enabled: (node: INode) => boolean
+
+	/**
+	 * Called by the files app if this tab has become the active tab or was deactivated.
+	 *
+	 * @param active - The new active state of this tab
+	 */
+	setActive: (active: boolean) => void | Promise<void>
+
+	/**
+	 * The lifecylce method for mounting the sidebar tab onto the sidebar.
+	 *
+	 * @param el - The element to mount the sidebar tab to
+	 * @param context - The current sidebar context
+	 */
+	mount: (el: HTMLElement, context: SidebarUpdateContext) => void | Promise<void>
+
+	/**
+	 * The lifecycle method for updating the sidebar tab.
+	 * This is called if the currently selected node changes.
+	 *
+	 * @param context - The current sidebar context
+	 */
+	update: (context: SidebarUpdateContext) => void | Promise<void>
+
+	/**
+	 * The lifecycle method for unmounting the sidebar tab.
+	 * This is called if the sidebar is unmounted from the files app and thus the sidebar tab needs to do its cleanup and unmounting.
+	 */
+	unmount: () => void | Promise<void>
+
+	/**
+	 * Called when the bottom of the sidebar was reached during scrolling.
+	 */
+	onScrollBottomReached?: () => void | Promise<void>
+}
+
+/**
+ * Register a new sidebar tab for the files app.
+ *
+ * @param tab - The sidebar tab to register
+ * @throws If the provided tab is not a valid sidebar tab and thus cannot be registered.
+ */
+export function registerSidebarTab(tab: ISidebarTab): void {
+	validateSidebarTab(tab)
+
+	window._nc_files_sidebar_tabs ??= new Map<string, ISidebarTab>()
+	if (window._nc_files_sidebar_tabs.has(tab.id)) {
+		logger.warn(`Sidebar tab with id "${tab.id}" already registered. Skipping.`)
+		return
+	}
+	window._nc_files_sidebar_tabs.set(tab.id, tab)
+	logger.debug(`New sidebar tab with id "${tab.id}" registered.`)
+}
+
+/**
+ * Get all currently registered sidebar tabs.
+ */
+export function getSidebarTabs(): ISidebarTab[] {
+	if (window._nc_files_sidebar_tabs) {
+		return [...window._nc_files_sidebar_tabs.values()]
+	}
+	return []
+}
+
+/**
+ * Check if a given sidebar tab objects implements all necessary fields.
+ *
+ * @param tab - The sidebar tab to validate
+ */
+function validateSidebarTab(tab: ISidebarTab): void {
+	if (typeof tab !== 'object') {
+		throw new Error('Sidebar tab is not an object')
+	}
+
+	if (!tab.id || (typeof tab.id !== 'string') || tab.id !== CSS.escape(tab.id)) {
+		throw new Error('Sidebar tabs need to have an id conforming to the HTML id attribute specifications')
+	}
+
+	if (!tab.displayName || typeof tab.displayName !== 'string') {
+		throw new Error('Sidebar tabs need to have a name set')
+	}
+
+	if (typeof tab.iconSvg !== 'string' || !tab.iconSvg.match(/^<svg.+<\/svg>$/i)) {
```

Comment:
We've been using is-svg everywhere else, let's stay consistent? :)
```suggestion
	if (typeof tab.iconSvg !== 'string' || !isSvg(tab.iconSvg))
```